src/dswx_sar/pre_processing.py

```diff
@@ -148,8 +190,9 @@ def _interpolate_gdal(self, ref_file,
                                resampleAlg=method,
                                format='GTIFF')
 
-        gdal.Warp(output_tif_str, input_tif_str, options=opt)
-
+        ds = gdal.Warp(output_tif_str, input_tif_str, options=opt)
+        ds = None
+        del ds
```

Comment:
I don't think `ds` survives outside the `_interpolate_gdal` function context.
```suggestion
        gdal.Warp(output_tif_str, input_tif_str, options=opt)
```